packages/core/src/types/transaction.ts

```diff
@@ -71,9 +71,9 @@ export type Transaction = Prettify<
         accessList?: never;
         gasPrice?: never;
         /** Total fee per gas in wei (gasPrice/baseFeePerGas + maxPriorityFeePerGas). Only present in EIP-1559 transactions. */
-        maxFeePerGas: bigint;
+        maxFeePerGas: bigint | undefined;
```

Comment:
Noticing that other fields in this type use the `?:` syntax, we should try to be consistent.